src/System.Management.Automation/engine/MshObject.cs

```diff
@@ -1875,16 +1864,16 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
             return note.Value;
         }
 
-        internal int GetSerializationDepth(TypeTable backupTypeTable)
+        internal int GetSerializationDepth(TypeTable? backupTypeTable)
         {
             int result = 0;
 
-            TypeTable typeTable = backupTypeTable ?? this.GetTypeTable();
+            TypeTable? typeTable = backupTypeTable ?? this.GetTypeTable();
             if (typeTable != null)
             {
-                PSMemberSet standardMemberSet = TypeTableGetMemberDelegate<PSMemberSet>(this,
+                PSMemberSet? standardMemberSet = TypeTableGetMemberDelegate<PSMemberSet>(this,
                     typeTable, TypeTable.PSStandardMembers);
-                result = (int)GetNoteSettingValue(standardMemberSet, TypeTable.SerializationDepth, 0, typeof(int), true, this);
+                result = (int)GetNoteSettingValue(standardMemberSet, TypeTable.SerializationDepth, 0, typeof(int), true, this)!;
```

Comment:
With my previous comment:
```suggestion
                result = (int)GetNoteSettingValue(standardMemberSet, TypeTable.SerializationDepth, 0, typeof(int), true, this);
```